hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/DecommissioningNodesWatcher.java

```diff
@@ -126,6 +127,11 @@ public void init(Configuration conf) {
         YarnConfiguration.RM_DECOMMISSIONING_NODES_WATCHER_POLL_INTERVAL,
         YarnConfiguration
           .DEFAULT_RM_DECOMMISSIONING_NODES_WATCHER_POLL_INTERVAL);
+    // expire interval should not be configured more than RM_AM_EXPIRY_INTERVAL_MS
+    this.expireIntvl = Math.min(conf.getLong(YarnConfiguration.RM_AM_EXPIRY_INTERVAL_MS,
```

Comment:
other time related confs like RM_AM_EXPIRY_INTERVAL_MS  and others in YarnConfiguration are added as milliseconds long/int instead of strings converted to duration so keeping similar for consistency. Let me know if changing to String duration is a must for going forward with this PR